runtime/altair/src/lib.rs

```diff
@@ -1302,6 +1303,89 @@ impl pallet_data_collector::Config for Runtime {
 	type Moment = Moment;
 }
 
+#[derive(Clone, Eq, Debug, PartialEq, Ord, PartialOrd, Encode, Decode, TypeInfo)]
+pub struct NullTransactor {}
+
+impl UtilityEncodeCall for NullTransactor {
+	fn encode_call(self, _call: UtilityAvailableCalls) -> Vec<u8> {
+		unimplemented!("XcmTransactor feature not used")
```

Comment:
When would one hit this logic? Just am worried about `unimplemented` in our logic.

Reply:
Good question! The only extrinsic from `xcm_transactor` we use is `pallet_xcm_transactor::Pallet::<T>::transact_through_sovereign(`, which doesn't use this config type:

```rust
		// XcmTransact needs to be implemented. This type needs to implement
		// utility call encoding and multilocation gathering
		type Transactor: Parameter + Member + Clone + XcmTransact;
```

`Transactor` is used in `transact_through_derivative`, and this extrinsic requires static knowledge on how to encode a `AsDerivative` call on a given destination chain. Again, we don't use this.

What is clear by going through this after your prompt is that we must make `xcm_transactor::Call` hidden to stop users from calling `transact_through_derivative` which would call this `unimplemented`. Thanks for the catch 👍

Reply:
@mustermeiszer actually just tested this scenario on our node on the Moonbase Alpha and `transact_through_derivative` fails early on because we'd need to register a derivative index account (`IndexToAccount` storage, which requires `EnsureRoot`).

So we _can_ make `impl UtilityEncodeCall` return `vec![]` but for `impl xcm_primitives::XcmTransact` which requires `fn destination(self) -> MultiLocation` we must return some value, for example `MultiLocation { 0, HERE }` which is a sort of a `dev/null` in this context.

The later is safer but in reality none of the options ever get hit. wdyt?

@wischli would be cool to get your thoughts on this as well

Reply:
I have DRYed the `NullTransactor` into `runtime_common` and made it a safer implementation: https://github.com/centrifuge/centrifuge-chain/pull/1490/commits/1e76620268ba07205882db02446b793d0954f497

I think this is a safer and cleaner approach for us to move forward with without loosing sleep :)

Reply:
> So we _can_ make `impl UtilityEncodeCall` return `vec![]` but for `impl xcm_primitives::XcmTransact` which requires `fn destination(self) -> MultiLocation` we must return some value, for example `MultiLocation { 0, HERE }` which is a sort of a `dev/null` in this context.
>
> The later is safer but in reality none of the options ever get hit. wdyt?

I would also feel much better going this path with a proper comment for both impls more or less citing your comment above. IMO, we should never add `unimplemented` or `todo` code to Altair/Centrifuge runtimes if not feature gated.

We should also add `transact_through_derivative` to the `BaseCallFilter` to ensure the call cannot be made. We cannot guarantee the calls behaviour will stay is forever. This at least guarantees, it cannot be called such that we don't need to rely on the assumption of erroring out earlier.

Reply:
Maybe lets add a note to the `construct_runtime!` section where we add the `pallet_xcm_transactor` that we never want to add the call to our runtime call unless someone takes care of the `Transactor`

Reply:
@wischli @mustermeiszer I forgot to correct something I said above; we do need the `xcmTransactor` `Call` to set some storage state. I agree that given we can't guarantee future behaviour for the xcm_transactor `transact_through_derivative` call we should just block it in the call filter.

Reply:
Done in https://github.com/centrifuge/centrifuge-chain/pull/1490/commits/3d1f7b8e546a7cc33da0f6eba68bf1d82ada1368